docs_test.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+exit_code=0
+
+DOCS_DIR="${TEST_SRCDIR}/org_tensorflow_tensorboard/docs/"
+ipynbs="$(find $DOCS_DIR -iname *.ipynb)"
+bad_files=()
+
+# Check whether ipynb have the right JSON format.
+for ipynb in $ipynbs; do
```

Comment:
This word-splitting is dangerous if there are ever spaces in the file
paths, which could occur if the user’s workspace has spaces in it (due
to `TEST_SRCDIR`). The unquoted expansion of `$DOCS_DIR` above is
likewise dangerous.

See <http://cl/276393831> for a recent occurrence of exactly this issue.

A safer way to do this is to save the output of `find` to a file and
iterate over its lines\* with `while read -r line`. (Usually you could
pipe the `find` directly into the `while`, but that creates a subshell,
and here you depend on modifications to `code` being propagated back
out.)

Or, see also suggestion below to consider using Python to avoid such
issues. :-)

\* This still assumes that no filename has *newlines* in it, but this is
generally a bit safer. If you want to be really robust you can use NUL
delimiters, which cannot appear in filenames.